docs/src/main/mdoc/typeclasses/bifoldable.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: docs
+title:  "Bifoldable"
+section: "typeclasses"
+source: "core/src/main/scala/cats/Bifoldable.scala"
+scaladoc: "#cats.Bifoldable"
+---
+
+# Bifoldable
+
+`Bifoldable` takes two type parameters and it's seen as a `Foldable` in both of these parameters.
+
+As a reminder `Foldable` is implemented in terms of `foldLeft` and `foldRight`, similarly `Bifoldable` is implemented in terms of:
+
+   ```scala
+   //eagerly performs a left-associative bi-fold over `fab` 
+    def bifoldLeft[A, B, C](fab: F[A, B], c: C)(f: (C, A) => C, g: (C, B) => C): C
+
+    //lazily performs a right-associative bi-fold over `fab` 
+    def bifoldRight[A, B, C](fab: F[A, B], c: Eval[C])(f: (A, Eval[C]) => Eval[C], g: (B, Eval[C]) => Eval[C]): Eval[C]
+   ```
+
+and by implementing those 2 methods you also get:
+   ```scala
+     def bifold[A, B](fab: F[A, B])(implicit A: Monoid[A], B: Monoid[B]): (A, B)
+  
+     def bifoldMap[A, B, C](fab: F[A, B])(f: A => C, g: B => C)(implicit C: Monoid[C]): C
+   ```
+
+## Either and Validated as Bifoldable
+
+Assume in your codebase multiple input paths and system requests that end up as `Either[Exception, *]` or `Validated[Exception, *]`.  
+We want to track the amount of errors and store the content of valid requests.
+
+Let's add the implicits:
+```scala mdoc
+import cats._
+import cats.data._
+import cats.implicits._
+```
+
+and define a summary class, capable of storing this info:
+```scala mdoc
+case class Report(content: List[String], errors: Int) {
+  def add(`new`: String): Report =
+    this.copy(content = content :+ `new`)
+
+  def reject(): Report =
+    this.copy(errors = errors + 1)
+}
+```
+
+`Bifoldable` is useful to get a summary value from `F[_,_]` data types:
+```scala mdoc
+def grow[T[_, _]: Bifoldable](current: Report)(one: T[_, String]): Report =
```

Comment:
Just ideas to make the names more specific :)
```suggestion
def updateReport[T[_, _]: Bifoldable](current: Report)(result: T[_, String]): Report =
```

Reply:
maybe just `update` ? 
I think the signature makes it self-explanatory